### PR TITLE
EE: "Static members", etc. should not inherit Access, StorageType, TypeModifierFlags from parent item

### PR DESCRIPTION
**Pull request by cston**

### CODE REVIEW

**Comment by cston** [comment]
@KevinH-MS, @amcasey, @pnelsonmsft, @ManishJayaswal

File: src/ExpressionEvaluator/CSharp/Test/ResultProvider/DebuggerDisplayAttributeTests.cs [hunk 1]
```diff
@@ -598,7 +598,7 @@ class B
 }
 ";
             DkmClrRuntimeInstance runtime = null;
-            GetMemberValueDelegate getMemberValue = (v, m) => (m == "Q") ? CreateErrorValue(runtime.GetType("A"), "Function evaluation timed out") : null;
+            GetMemberValueDelegate getMemberValue = (v, n, t, p, i) => (n == "Q") ? CreateErrorValue(runtime.GetType("A"), "Function evaluation timed out") : null;
```
Comment (amcasey) [inline src/ExpressionEvaluator/CSharp/Test/ResultProvider/DebuggerDisplayAttributeTests.cs]: I find these single-letter parameter names hard to read/understand.

File: src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/EvalResultDataItem.cs [hunk 2]
```diff
@@ -19,7 +19,7 @@ internal enum ExpansionKind
         RawView,
         ResultsView,
         StaticMembers,
-        TypeVariables
+        TypeVariable
```
Comment (amcasey) [inline src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/EvalResultDataItem.cs]: Isn't it called "TypeVariables" in the UI?

**Comment by amcasey** [comment]
I'm having some trouble figuring out which changes to review - it looks like this PR includes some merge commits.

File: src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/EvalResultDataItem.cs (hunk 2 above)
Comment (cston) [inline src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/EvalResultDataItem.cs]: This represents one of the TypeVariables items.

File: src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.cs [hunk 3]
```diff
@@ -133,36 +133,120 @@ private void CreateEvaluationResultAndContinue(EvalResultDataItem dataItem, Work
                         DataItem: null));
                     break;
                 case ExpansionKind.NativeView:
+                    {
```
Comment (KevinH-MS) [inline src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.cs]: remove extra '{', '}' around this block?

**Comment by KevinH-MS** [comment]
LGTM

File: src/ExpressionEvaluator/CSharp/Test/ResultProvider/DebuggerDisplayAttributeTests.cs (hunk 1 above)
Comment (cston) [inline src/ExpressionEvaluator/CSharp/Test/ResultProvider/DebuggerDisplayAttributeTests.cs]: I've reverted this change.

File: src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.cs (hunk 3 above)
Comment (cston) [inline src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.cs]: I'd like to ensure the names are scoped to this block.

**Pull request merged**